lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +199,40 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self._seed_urls = config.get_seed_urls()
+        self._config = config
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        url = article_bs['href']
+        if isinstance(url, str):
+            return url
+        return url[0]
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        for seed_url in self._seed_urls:
+            res = make_request(seed_url, self._config)
+            soup = BeautifulSoup(res.content, "lxml")
+            for paragraph in soup.find_all('a', class_="article-list__title"):
+                if len(self.urls) >= self._config.get_num_articles():
+                    return
+                try:
+                    url_ = self._extract_url(paragraph)
+                except KeyError:
```

Comment:
I do not see reason to let exception be handled here instead of safe retrieval of value by a key in a nested function